src/main/scala/com/kinja/amqp/AmqpConsumer.scala

```diff
@@ -115,6 +122,10 @@ class AmqpConsumer(
 							logger.warn(s"""[RabbitMQ] Couldn't parse message "$s" : $e""")
 							sender ! Reject(envelope.getDeliveryTag, requeue = false)
 					}
+				case WakeUp => ()
+				case Disconnect =>
+					context.become(disconnected)
```

Comment:
I'm wondering whether this (and the implementation is the previous iteration) would actually address the problem of shutting down the service in the middle of processing the message. They I see this works now:
1. The service is running and the consumer receives a `Delivery` message, so it starts processing it within the `Await.result`.
2. While processing is happening, ops wants to shut down the service, so they send a `Disconnect` message. This message will be appended to the end of the actor message queue and won't be processed until all preceding messages are processed too.
3. Ops doesn't wait for the `Disconnect` message to be processed and after sending the message they assume RabbitMQ messages have been terminated. But that's not the case since processing of `Delivery` is still in progress and the `Disconnect` message is only enqueued.

So in order to fix this, the RabbitMQ client needs to send back a signal when disconnecting is complete, and ops (the `disableService` API call) must await the receipt of this acknowledgement.

Reply:
Hmmm, we'll have 30 seconds delay between `disableService` and killing. But it does require some digging.

Reply:
I guess that's not enough if processing a message takes 2 minutes. 30 seconds is just an arbitrary number.

Reply:
If processing a message takes 2 minutes, we're screwed anyway. I'm more concerned about what happens if there are 100 messages, each taking a second to process.